game/scripts/vscripts/components/boss/ngp.lua

```diff
@@ -17,50 +22,115 @@ function NGP:Init ()
   })
 
   CustomGameEventManager:RegisterListener('ngp_selection', Dynamic_Wrap(NGP, 'PlayerVote'))
+
+
+  for playerId = 0,19 do
+    local player = PlayerResource:GetPlayer(playerId)
+    if player ~= nil then
+      if player:GetTeam() == 3 then
+        totalbadplayers = totalbadplayers + 1
+      elseif player:GetTeam() == 2 then
+        totalgoodplayers = totalgoodplayers + 1
+      end
+    end
+  end
 end
 
 function NGP:PlayerVote (eventSourceIndex, args)
-  DebugPrintTable(eventSourceIndex)
+  getGlobalTable(team) 
+  -- DebugPrintTable(eventSourceIndex)
   -- DebugPrintTable(args)
   local playerID = eventSourceIndex.PlayerID
   local id = eventSourceIndex.id
   local option = eventSourceIndex.option
   local team = ShortTeamName(PlayerResource:GetTeam(playerID))
   local item = NGP.activeItems[tonumber(id)]
-
+  local heroname = PlayerResource:GetSelectedHeroName(playerID)
   if item.team ~= team then
     Notifications:TopToAll({text="NGP mismatch " .. item.team .. " vs " .. team, duration=2.0})
     return
   end
-
-  DebugPrint(team)
   item.votes[playerID] = option
+  item.heroname[playerID] = heroname
+
+  setTableItem(item, team)
+  
+  if item.finished == false then
+    local totalvoted = 0
+    for i = 0,19 do
+      if item.votes[i] then
+        totalvoted = totalvoted + 1
+      end
+    end
+    if totalvoted > (totalbadplayers-1) and team == "bad" then
+      Timers:RemoveTimer(NGP.activeTimers[tonumber(id)])
+      NGP:FinishVoting(tonumber(id), team)
+    elseif totalvoted > (totalgoodplayers-1) and team == "good" then
+      Timers:RemoveTimer(NGP.activeTimers[tonumber(id)])
+      NGP:FinishVoting(tonumber(id), team)
+    end
+  end
+
+  --getGlobalTable(team)
+  --DebugPrintTable(ngpItemsGood)
+  --DebugPrint("sssss")
+
 end
 
 function NGP:GiveItemToTeam (item, team)
-  local ngpItems = CustomNetTables:GetTableValue('ngp', team)
+  getGlobalTable(team)
 
   DebugPrint('item index will be ' .. NGP.itemIndex)
-  DebugPrintTable(ngpItems)
   item.id = NGP.itemIndex
+  item.finished = false
   NGP.itemIndex = NGP.itemIndex + 1
-
   item.team = team
   NGP.activeItems[item.id] = item
-
   NGP.activeItems[item.id].votes = {}
+  NGP.activeItems[item.id].heroname = {}
 
+  setTableItem(item, team)
 
-  ngpItems[item.id] = item
+  NGP.activeTimers[item.id] = Timers:CreateTimer(60, function ()
+    NGP:FinishVoting(item.id, team)
+  end)
+end
 
-  CustomNetTables:SetTableValue('ngp', team, ngpItems)
 
-  Timers:CreateTimer(60, function ()
-    NGP:FinishVoting(NGP.activeItems[item.id])
-  end)
+function getGlobalTable(team) 
```

Comment:
`getGlobalTable` and the other net table functions here should be within the NGP class. i.e. declare as `function NGP:getGlobalTable`. Otherwise the functions will be in global scope.

You would then call them with `self:getGlobalTable`, etc.